lib/Steps/Integrations/NextJs.ts

```diff
@@ -303,6 +321,48 @@ export class NextJs extends BaseIntegration {
     fs.writeFileSync(targetPath, filledTemplate);
   }
 
+  private _hasPackageInstalled(packageName: string): boolean {
+    const depsVersion = _.get(appPackage, ['dependencies', packageName]);
+    const devDepsVersion = _.get(appPackage, ['devDependencies', packageName]);
+    return !!depsVersion || !!devDepsVersion;
+  }
+
+  private _getPackageMangerChoice(): PackageManager | null {
+    if (fs.existsSync(path.join(process.cwd(), 'yarn.lock'))) {
+      return 'yarn';
+    }
+    if (fs.existsSync(path.join(process.cwd(), 'pnpm-lock.yaml'))) {
+      return 'pnpm';
+    }
+    if (fs.existsSync(path.join(process.cwd(), 'package-lock.json'))) {
+      return 'npm';
+    }
+    return null;
+  }
+
+  private _getInstallCommand(packageManager: PackageManager): string {
+    switch (packageManager) {
+      case 'yarn':
+        return 'yarn add';
+      case 'pnpm':
+        return 'pnpm add';
+      case 'npm':
+        return 'npm install';
+      default:
+        throw new Error(`Unknown package manager: ${packageManager}`);
+    }
+  }
+
+  private async _installPackage(
+    packageName: string,
+    packageManager: PackageManager,
+  ): Promise<void> {
+    const command = this._getInstallCommand(packageManager);
+    await promisify(exec)(`${command} ${packageName}`);
```

Comment:
L: Any reason not to just use `execSync` here?

Reply:
@lobsterkatie In case you want to parallelize it and have it run faster